### PR TITLE
fix person modal across multiple dates

### LINKED ISSUE #5818: Person modal data consistency issues
## Bug description

When I create a fairly simple query (event + 1 filter) and click on the graph to open the person models (sep 3), I get 25 people, whereas the graph says there should be 17 ([see this internal link](https://app.posthog.com/insights?insight=TRENDS&interval=day&display=ActionsLineGraph&events=%5B%7B%22id%22%3A%22insight%20loaded%22%2C%22name%22%3A%22insight%20loaded%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22math%22%3A%22dau%22%7D%5D&filter_test_accounts=true&actions=%5B%5D&new_entity=%5B%5D&breakdown=false&breakdown_type&properties=%5B%7B%22key%22%3A%22insight%22%2C%22value%22%3A%5B%22STICKINESS%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%5D&date_from=2021-08-30&date_to=2021-09-06#personModal=%7B%22action%22%3A%7B%22id%22%3A%22insight%20loaded%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22name%22%3A%22insight%20loaded%22%2C%22math%22%3A%22dau%22%2C%22math_property%22%3Anull%2C%22properties%22%3A%5B%5D%7D%2C%22label%22%3A%22insight%20loaded%22%2C%22date_from%22%3A%222021-09-03%22%2C%22date_to%22%3A%222021-09-03%22%2C%22filters%22%3A%7B%22insight%22%3A%22TRENDS%22%2C%22interval%22%3A%22day%22%2C%22display%22%3A%22ActionsLineGraph%22%2C%22events%22%3A%5B%7B%22id%22%3A%22insight%20loaded%22%2C%22name%22%3A%22insight%20loaded%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22math%22%3A%22dau%22%7D%5D%2C%22filter_test_accounts%22%3Atrue%2C%22actions%22%3A%5B%5D%2C%22new_entity%22%3A%5B%5D%2C%22breakdown%22%3Afalse%2C%22breakdown_type%22%3Anull%2C%22properties%22%3A%5B%7B%22key%22%3A%22insight%22%2C%22value%22%3A%5B%22STICKINESS%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%5D%2C%22date_from%22%3A%222021-08-30%22%2C%22date_to%22%3A%222021-09-06%22%7D%2C%22saveOriginal%22%3Atrue%7D))

The same thing happens with a [breakdown query](https://app.posthog.com/insights?insight=TRENDS&interval=day&display=ActionsLineGraph&events=%5B%7B%22id%22%3A%22insight%20loaded%22%2C%22name%22%3A%22insight%20loaded%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22math%22%3A%22dau%22%7D%5D&filter_test_accounts=true&properties=%5B%5D&actions=%5B%5D&new_entity=%5B%5D&breakdown=insight&breakdown_type=event)

## Expected behavior

The person modal should have 17 people

## How to reproduce

1. Go to link above
2. Click on sep 3
3. Verify there are 25 people

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: _please provide_

## Additional context

Looks like when passing through the date as 3rd of september, the query looks like this, which is probably where the 25 comes from (3rd + 4th of sept combined):
image

cc @EDsCODE @macobo

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Closes #5818 

I'm not 100% _why_ we need to do this or why this has happened (or whether it has been like this for a long time). @EDsCODE any thoughts?

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by EDsCODE** [comment]
We merged [this](https://github.com/PostHog/posthog/pull/5572) a few weeks ago trying to fix a bug on trends cumulative people. I think the error is that I set the `+ timedelta(days=1)` without noticing that on line 144 in ee/clickhouse/views/actions.py parse_timestamps will already substitute the timestamp to be the end of the day.

**Comment by EDsCODE** [comment]
Going to consolidate the date-time handling in a better way so there's not overlapping concerns

File: posthog/api/test/test_action_people.py [hunk 1]
```diff
@@ -327,18 +341,72 @@ def test_day_interval(self):
...
+            )
+            event_factory(
+                team=self.team, event="sign up", distinct_id="person2", timestamp="2020-01-04T20:00:00Z",
+            )
+            outside_range_person = person_factory(team_id=self.team.pk, distinct_ids=["outside_range"])
+            event_factory(
+                team=self.team, event="sign up", distinct_id="outside_range", timestamp="2020-01-02T13:50:00Z",
+            )
+            event_factory(
+                team=self.team, event="sign up", distinct_id="outside_range", timestamp="2020-01-05T15:50:00Z",
```
Comment (EDsCODE) [inline posthog/api/test/test_action_people.py:366]: I think these are the dates to test for?

Comment (timgl) [reply posthog/api/test/test_action_people.py:366]: this test was actually fine I think. The failure (without my fix) was caused by lines 317-323

**Comment by macobo** [comment]
I think we've juggled the date filtering system around before, so seems like a likely candidate to break. Long-term we should avoid having one-off date manipulation fns like this.

Merging, nice fix!

**Pull request merged**